alicloud/resource_alicloud_emr_cluster.go

```diff
@@ -0,0 +1,456 @@
+)
+
+func resourceAlicloudEmrCluster() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAlicloudEmrClusterCreate,
+		Read:   resourceAlicloudEmrClusterRead,
+		Update: resourceAlicloudEmrClusterUpdate,
+		Delete: resourceAlicloudEmrClusterDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"cluster_type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"emr_ver": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"charge_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"host_group": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				ForceNew: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"host_group_name": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"host_group_type": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"charge_type": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"period": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"node_count": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"instance_type": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"disk_type": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"disk_capacity": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"disk_count": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"sys_disk_type": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"sys_disk_capacity": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"auto_renew": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"vswitch_id": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"gpu_driver": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+			},
+			"zone_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"security_group_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"is_open_public_ip": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				ForceNew: true,
+			},
+			"security_group_name": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"net_type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"vpc_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+			"vswitch_id": {
```

Comment:
> I think `net_type` and `vpc_id` can be removed and them can be got by `vswitch_id`. If `vswitch` is set, the network type should be `vpc` and vpc_id can be fetched by `vswithc_id`. If not, the network type should not `vpc`

You are right. But because of the api design and implementation, it does not take the mapping relationship into account. 
Maybe I can persuade the owner to do api improvement later.

Reply:
Other products API also has this problems and they also have removed the `net_type` and `vpc_id`